### DISCUSSION TITLE
Session returns NULL in middleware after logging in with Auth-UI on Next.js.

### DISCUSSION THREAD

**Discussion by jmcannon** [supabase/supabase#13535] [Questions]
I'm successfully logging in with Auth-UI - the session is created on the backend, the cookie is saved, and I can get the User from useUser(). 

However, I'm using the boilerplate middleware code to require auth for certain paths (https://supabase.com/docs/guides/auth/auth-helpers/nextjs#auth-with-nextjs-middleware) but getSession always returns null, and therefore continues to redirect everything back to the login page.

Am I missing a basic concept? I assumed Auth-UI wrote the auth-token to a cookie, and my middleware client would immediately be able to read that cookie. Am I missing a step? All my code is boilerplate from the Next.js tutorials in the docs.

This error is also being written to the terminal: Failed to parse cookie string: [Error: The edge runtime does not support Node.js 'buffer' module]. It's confusing to me, because I'm not using the Edge runtime.

Any help is very much appreciated.

**Comment by BrunoFerreira95** [comment]
Hello, I to fix this issue you need to understand, with the new app router, we need to use the “use client”. without makes the return session null in the terminal.

This made the page of session work on the server side. To make work on browser, you need to implement the “use client” to make this works on browser. 

'use client'
let session
  useEffect(() => {
    const initSession = async () => {
      const { data, error } = await supabase.auth.getSession()
      session = data
    }    
    initSession()
  })
